### PR TITLE
PR for Authentication

### PR DESCRIPTION
**Pull request by SK02K1**
# Authentication added

### Features

- Login.
- Signup.
- Logout.
- RequiresAuth component to validate private routes.
- Storing userData in local storage on login and signup for future refrence.
- Error messages on forms

Screenshot 2022-05-16 at 10 46 20 PM

Screenshot 2022-05-16 at 10 46 29 PM

Screenshot 2022-05-16 at 10 48 05 PM

Screenshot 2022-05-16 at 10 48 21 PM

Screenshot 2022-05-16 at 10 49 51 PM

### CODE REVIEW

File: src/components/Forms/SignupForm.jsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  SimpleGrid,
+  useColorModeValue,
+  Link,
+  VStack,
+  Spinner,
+  FormErrorMessage,
+} from '@chakra-ui/react';
+import { useState } from 'react';
+import { Link as RouterLink, useLocation, Navigate } from 'react-router-dom';
+import { inputChangeHandler } from '../../utilities';
```
Comment (Rohitprasad83) [inline src/components/Forms/SignupForm.jsx]: can use absolute path https://stackoverflow.com/questions/41993461/import-react-components-with-absolute-path

**Review by Rohitprasad83 — APPROVED** [review]
Code looks good, just added one suggestion.

File: src/components/Forms/SignupForm.jsx (hunk 1 above)
Comment (SK02K1) [reply src/components/Forms/SignupForm.jsx]: @Rohitprasad83 Thanks for the suggestion 🙌🏻

**Review by ragini07 — COMMENTED** [review]
Good folder structure
Clean code

**Pull request merged**